<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!-- Arguments -->
  <arg name="initial_ts_state_from_agent" default="False"/>
<!--   <arg name="num_mobile" default=1/> -->
<!--   <arg name="num_quadruped" default=1/> -->

  <arg name="agent_name_mobile_1" default="dr_0"/>
<!--  <arg name="agent_name_mobile_2" default="openshelf_1"/>-->
  <arg name="agent_name_dog_1" default="a1_gazebo"/>
  <arg name="rviz" default="true"/>

<!--   <param name="agent_name" type="string" value="$(arg agent_name)" /> -->
  <!-- hard and soft task parameters -->
  <rosparam command="load" file="$(find ltl_automaton_planner)/config/ltl_formula_final_1.yaml" />

  <!-- Transition system for mobile robot-->
  <param name="transition_system_mobile_1_textfile" type="str" textfile="$(find ltl_automaton_planner)/config/ts_delivery.yaml" />

  <!-- Transition system for mobile robot-->
  <param name="transition_system_mobile_textfile" type="str" textfile="$(find ltl_automaton_planner)/config/ts_delivery.yaml" />

  <!-- Transition system for quadruped-->
  <param name="transition_system_quadruped_textfile" type="str" textfile="$(find ltl_automaton_planner)/config/ts_quadruped.yaml" />

  <!-- Run planner node -->
<!--   <node name="ltl_planner" pkg="ltl_automaton_planner" type="multi_robot_planner_node.py" output="screen"> -->
<!--     <param name="initial_ts_state_from_agent" type="bool" value="$(arg initial_ts_state_from_agent)" /> -->
<!--   </node> -->

  <!-- Run the gazebo world-->
  <include file="$(find unitree_bringup)/launch/gazebo.launch">
    <arg name="world_name" value="hospital_2"/>
    <arg name="world_format" value="world"/>
  </include>

  <include file="$(find ltl_automation_a1)/launch/dr_monitor_bt.launch">
    <arg name="agent_name" value="$(arg agent_name_mobile_1)" />
    <arg name="plan_name" value="/action_plan_1" />
    <arg name="pose_x" value="-21" />
    <arg name="pose_y" value="29.4" />
  </include>

  <include file="$(find ltl_automation_a1)/launch/a1_monitor_bt.launch">
    <arg name="agent_name" value="$(arg agent_name_dog_1)" />
    <arg name="plan_name" value="/action_plan_2" />
    <arg name="pose_x" value="-8.2" />
    <arg name="pose_y" value="29.0" />
    <arg name="init_yaw" value="3.14" />
  </include>


  <node if="$(arg rviz)" name="rviz" pkg="rviz" type="rviz"
        args="-d $(find unitree_navigate)/rviz/multi_robot_mocap.rviz -f map"
        output="screen"/>


</launch>
